transforms v2 rotate can create different sizes between tensors and PIL with expand=True

```python
import torch
from torchvision.transforms.v2 import functional as F

input_tensor = torch.rand(3, 17, 11)
input_pil = F.to_pil_image(input_tensor)

kwargs = dict(angle=-10.9, center=(1.2, 4.9), expand=True)

output_tensor = F.rotate(input_tensor, **kwargs)
output_pil = F.rotate(input_pil, **kwargs)

shape_tensor = F.get_image_size(output_tensor)
shape_pil = F.get_image_size(output_pil)

assert shape_tensor == shape_pil, f"{shape_tensor} != {shape_pil}"
```

```
AssertionError: [15, 20] != [15, 19]
```

Replacing the import with `from torchvision.transforms import functional as F`, i.e. using transforms v1, does not have this problem.

PIL kernel is the culprit:

```python
import torch
from torchvision.transforms.v2 import functional as F

input_tensor = torch.rand(3, 17, 11)
input_pil = F.to_pil_image(input_tensor)

kwargs = dict(angle=-10.9, expand=True)

output_w_center = F.rotate(input_tensor, **kwargs, center=(1.2, 4.9))
output_wo_center = F.rotate(input_pil, **kwargs, center=None)

shape_w_center = F.get_image_size(output_w_center)
shape_wo_center = F.get_image_size(output_wo_center)

assert shape_w_center == shape_wo_center, f"{shape_w_center} != {shape_wo_center}"
```

```
AssertionError: [15, 20] != [15, 19]
```

Meaning, #7271 did its job, but we need to expand it to the PIL kernel as well.

The issue is that `center` theoretically should have no effect on the output size if `expand=True` is given. Practically, rounding effects take place which lead the off-by-one errors. The difference between v1 and v2 is that we not only warn users about setting both parameters

https://github.com/pytorch/vision/blob/71968bc4afb8892284844a7c4cbd772696f42a88/torchvision/transforms/v2/functional/_geometry.py#L922-L923

we also explicitly unset `center` in this case

https://github.com/pytorch/vision/blob/71968bc4afb8892284844a7c4cbd772696f42a88/torchvision/transforms/v2/functional/_geometry.py#L924

--- a/torchvision/transforms/v2/functional/_geometry.py
+++ b/torchvision/transforms/v2/functional/_geometry.py
@@ -921,7 +921,6 @@ def rotate_image_pil(
 
     if center is not None and expand:
         warnings.warn("The provided center argument has no effect on the result if expand is True")
-        center = None
 
     return _FP.rotate(
         image, angle, interpolation=pil_modes_mapping[interpolation], expand=expand, fill=fill, center=center
@@ -938,7 +937,6 @@ def rotate_bounding_box(
 ) -> Tuple[torch.Tensor, Tuple[int, int]]:
     if center is not None and expand:
         warnings.warn("The provided center argument has no effect on the result if expand is True")
-        center = None
 
     return _affine_bounding_box_with_expand(
         bounding_box,

